File: back-end/src/main/java/dev/davidsilva/music/playlist/PlaylistItemDto.java
package dev.davidsilva.music.playlist;

import com.fasterxml.jackson.annotation.JsonInclude;
import dev.davidsilva.music.song.SongDto;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@JsonInclude(JsonInclude.Include.NON_DEFAULT)
public class PlaylistItemDto implements Serializable {
    private Integer id;
    private Integer position;
    private String mbTrackId;
    private String songTitle;
    private String mbAlbumId;
    private String albumTitle;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private SongDto song;
    // Will not be adding a "playlist" field to this DTO, because ItemDtos
    // are never supposed to be handled independently, by themselves.
    // The way to create/read/update/delete a PlaylistItem is through the
    // corresponding PlaylistDto.
}

File: front-end/src/app/playlists/models/playlist-item-dto.ts
import type { SongDto } from '../../songs/song-dto';

export interface PlaylistItemDto {
  id: number;
  position: number;
  song: SongDto;
  // I think for now I'm not going to use
  // the cached properties in the front-end,
  // since we should always have a song
  mbTrackId?: string;
  songTitle?: string;
  mbAlbumId?: string;
  albumTitle?: string;
}
